src/lib/drivers/smbus/SMBus.cpp

```diff
@@ -154,6 +173,9 @@ int SMBus::block_write(const uint8_t cmd_code, void *data, uint8_t byte_count, b
 
 uint8_t SMBus::get_pec(uint8_t *buff, const uint8_t len)
 {
+#ifdef __NUTTX
+	return crc8ccitt(buff, len);
```

Comment:
Why bother added an ifdef for this now? Unless this fixes something I'd rather avoid platform defines and take a separate pass to refactor this into a new lib. https://github.com/PX4/Firmware/issues/15378

Reply:
The current CRC function is so non efficient. it 2 cascade loops, with shifts, xor and so on.
so it really feels like a crime to keep current implantation. 
does the idea of copying the nuttx's crc8ccitt() function to a px4 CRC lib, sounds good? is it ok 'hierarchy'?

Reply:
Yes pulling what we need into PX4 sounds good. I also agree about the efficiency, but I'd rather leave it as is for motivation to do the proper fix everywhere.

Reply:
I reverted that change. and will work on cross platform lib on other pr